jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1488,13 +1491,21 @@ Method getMethodHelper(
 			return null;
 		}
 	}
-	result = forDeclaredMethod ? getDeclaredMethodImpl(name, parameterTypes, strSig, null) : getMethodImpl(name, parameterTypes, strSig);
+	
+	if (forDeclaredMethod) {
+		result = getDeclaredMethodImpl(name, parameterTypes, strSig, null);
+	} else {
+		result = getMethodImpl(name, parameterTypes, strSig);
+		/* Retrieve the specified method implemented by the superclass from the top to the bottom */
+		resultFromSuperClass = this.isInterface() ? null : getMethodImplBySuperClass(this, name, strSig, parameterTypes);
+	}
+	
 	if (result == null) {
 		return throwExceptionOrReturnNull(throwException, name, parameterTypes);
 	}
 	if (0 == Reflection.filterMethods(this, new Method[] {result}).length) {
 		return throwExceptionOrReturnNull(throwException, name, parameterTypes);
-	}	
+	}
 
 	/*[PR 127079] Use declaring classloader for Methods */
 	/*[PR CMVC 104523] ensure parameter types are visible in the receiver's class loader */
```

Comment:
Should the parameter types of ` resultFromSuperClass` be checked visibility in the receiver's classcloaer as well?

Reply:
As discussed with @JasonFengJ9 ,  the check should be added as the `classloader1 `(loading the interface with specified methodfrom the superclass) might be different from the `classloader2 `(loading the interface with specified method from the current class).

The check was added and verified with the failing tests.

Reply:
I think only the actual returning method and the methods added into `methodList` are required for visibility check, so only one of `result` or `resultFromSuperClass` need to be checked, i.e., there is no need to check both.

Reply:
Updated the code and verified with the failing test cases.